packages/ui/pages/api/apyData.ts

```diff
@@ -0,0 +1,54 @@
+import { createClient } from '@supabase/supabase-js';
+import { config } from '@ui/config/index';
+import type { NextApiRequest, NextApiResponse } from 'next';
+
+const handler = async (req: NextApiRequest, res: NextApiResponse) => {
+  const { underlyingAddress, pluginAddress, rewardAddress } = req.query;
+
+  const client = createClient(config.supabaseUrl, config.supabasePublicKey);
+  let pricePerShares;
+
+  if (rewardAddress) {
+    pricePerShares = await client
+      .from('apy_flywheel')
+      .select('pricePerShare,created_at')
+      .eq('pluginAddress', (pluginAddress as string).toLowerCase())
+      .eq('rewardAddress', (rewardAddress as string).toLowerCase())
+      .eq('underlyingAddress', (underlyingAddress as string).toLowerCase())
+      .order('created_at', { ascending: false })
+      .limit(2);
```

Comment:
We probably will want to filter based on time ranges rather than number of entries